kernel/host/xmicterm/host/xmicterm.cc

```diff
@@ -129,16 +144,73 @@ class file_logger {
     std::unordered_map<uint16_t, std::unique_ptr<std::ofstream>> files;
 };
 
+/// Wrapper to the micctrl application, that controls the xeon phi coprocessor
+class mic_ctrl {
+public:
+    mic_ctrl(int adapter_)
+        :adapter(adapter_) {
+      }
+
+
+    void boot() {
+      std::stringstream ss;
+      ss << "sudo sh -c \"echo \\\"boot:elf:`pwd`/boot64.elf\\\" > /sys/class/mic/mic"
```

Comment:
It would be nice to pass the (absolute?) path to the kernel image as argument to the xmicterm program.